packages/core/primitives/event-dispatch/src/dispatcher.ts

```diff
@@ -6,81 +6,54 @@
+ */
+export type Replayer = (eventInfoWrappers: EventInfoWrapper[]) => void;
 
 /**
- * A global handler is dispatched to before normal handler dispatch. Returning
- * false will `preventDefault` on the event.
+ * A handler is dispatched to during normal handling.
  */
-export type GlobalHandler = (event: Event) => boolean | void;
+export type EventInfoHandler = (eventInfoWrapper: EventInfoWrapper) => void;
 
 /**
- * A replayer is a function that is called when there are queued events,
- * either from the `EventContract` or when there are no detected handlers.
+ * A global handler is dispatched to before normal handler dispatch. Returning
+ * false will `preventDefault` on the event.
  */
-export type Replayer = (eventInfoWrappers: EventInfoWrapper[], dispatcher: Dispatcher) => void;
 
+export type GlobalHandler = (event: Event) => boolean | void;
 /**
  * Receives a DOM event, determines the jsaction associated with the source
  * element of the DOM event, and invokes the handler associated with the
  * jsaction.
  */
 export class Dispatcher {
-  private readonly baseDispatcher: BaseDispatcher;
-
-  /** Whether to stop propagation for an `EventInfo`. */
-  private readonly stopPropagation: boolean;
-
-  /**
-   * The actions that are registered for this Dispatcher instance.
-   * This should be the primary one used once migration off of registerHandlers
-   * is done.
-   */
-  private readonly actions: {[key: string]: EventInfoWrapperHandler} = {};
-
-  /** A map of global event handlers, where each key is an event type. */
-  private readonly globalHandlers = new Map<string, Set<GlobalHandler>>();
-
-  /** The event replayer. */
+  /** The queue of events. */
+  private readonly queuedEventInfoWrappers: EventInfoWrapper[] = [];
+  /** The replayer function to be called when there are queued events. */
   private eventReplayer?: Replayer;
+  /** Whether the event replay is scheduled. */
+  private eventReplayScheduled = false;
 
   /**
-   * Receives a DOM event, determines the jsaction associated with the source
-   * element of the DOM event, and invokes the handler associated with the
-   * jsaction.
-   *
-   * @param getHandler A function that knows how to get the handler for a
-   *     given event info.
+   * Options are:
+   *   1. `eventReplayer`: When the event contract dispatches replay events
+   *      to the Dispatcher, the Dispatcher collects them and in the next tick
+   *      dispatches them to the `eventReplayer`.
+   * @param dispatchDelegate A function that should handle dispatching an `EventInfoWrapper` to handlers.
    */
   constructor(
-    private readonly getHandler?: (
-      eventInfoWrapper: EventInfoWrapper,
-    ) => EventInfoWrapperHandler | void,
-    {
-      stopPropagation = false,
-      eventReplayer = undefined,
-    }: {stopPropagation?: boolean; eventReplayer?: Replayer} = {},
+    private readonly dispatchDelegate: (eventInfoWrapper: EventInfoWrapper) => void,
```

Comment:
Can we use `EventInfoHandler` here?
```suggestion
    private readonly dispatchDelegate: EventInfoHandler,
```

Reply:
We could but my plan is to rename EventInfoHandler to EventInfoWrapperHandler and move it to LegacyDispatcher, as some people in g3 still use it.

In general, I don't find these utility types to be particularly useful, the only reason I've preserved it is because usages of that type were formerly exported.

Will leave it slated for deletion unless you ask again

Reply:
The plan to move this to the legacy dispatcher sounds good (no need to change anything in this PR) 👍